### PR TITLE
2022-p1: update buffer pool manager code documentation based on code review

### PR DESCRIPTION
**Pull request by timlee0119**

### CODE REVIEW

**Review by skyzh — APPROVED** [review]

File: src/include/buffer/buffer_pool_manager_instance.h [hunk 1]
```diff
@@ -56,36 +56,52 @@ class BufferPoolManagerInstance : public BufferPoolManager {
...
+  /**
+   * TODO(P1): Add implementation
+   *
+   * @brief Fetch the requested page from the buffer pool. Return nullptr if page_id needs to be fetched from the disk
+   * but all other frames are currently pinned.
+   *
+   * First search for page_id in the buffer pool. If not found, pick a replacement frame from either the free list or
+   * the replacer (always find from the free list first), read the page from disk by calling disk_manager_->ReadPage(),
+   * and replace the old page in the frame. Similar to NewPgImp(), you need to write the old dirty page back to disk,
+   * update the metadata of the new page
```
Comment (thepinetree) [inline src/include/buffer/buffer_pool_manager_instance.h:84]:
```suggestion
   *  to disk and update the metadata of the new page
```

File: src/include/buffer/buffer_pool_manager_instance.h [hunk 2]
```diff
@@ -56,36 +56,52 @@ class BufferPoolManagerInstance : public BufferPoolManager {
...
+
+  /**
+   * TODO(P1): Add implementation
+   *
+   * @brief Fetch the requested page from the buffer pool. Return nullptr if page_id needs to be fetched from the disk
+   * but all other frames are currently pinned.
+   *
+   * First search for page_id in the buffer pool. If not found, pick a replacement frame from either the free list or
+   * the replacer (always find from the free list first), read the page from disk by calling disk_manager_->ReadPage(),
+   * and replace the old page in the frame. Similar to NewPgImp(), you need to write the old dirty page back to disk,
```
Comment (thepinetree) [inline src/include/buffer/buffer_pool_manager_instance.h:83]:
```suggestion
   * and replace the old page in the frame. Similar to NewPgImp(), if the old page is dirty, you need to write it back
```

File: src/include/buffer/buffer_pool_manager_instance.h [hunk 3]
```diff
@@ -56,36 +56,52 @@ class BufferPoolManagerInstance : public BufferPoolManager {
...
+   * @param[out] page_id id of created page
+   * @return nullptr if no new pages could be created, otherwise pointer to new page
+   */
+  auto NewPgImp(page_id_t *page_id) -> Page * override;
+
+  /**
+   * TODO(P1): Add implementation
+   *
+   * @brief Fetch the requested page from the buffer pool. Return nullptr if page_id needs to be fetched from the disk
+   * but all other frames are currently pinned.
```
Comment (thepinetree) [inline src/include/buffer/buffer_pool_manager_instance.h:79]:
```suggestion
   * but all frames are currently in use and not evictable (in another word, pinned).
```

File: src/include/buffer/buffer_pool_manager_instance.h [hunk 4]
```diff
@@ -98,65 +114,32 @@ class BufferPoolManagerInstance : public BufferPoolManager {
...
    */
-  auto NewPgImp(page_id_t *page_id) -> Page * override;
+  void FlushAllPgsImp() override;
 
   /**
    * TODO(P1): Add implementation
    *
-   * @brief Delete a page from the buffer pool.
+   * @brief Delete a page from the buffer pool. If page_id is not in the buffer pool, do nothing and return true. If the
+   * pin count of page_id is greater than 0, don't delete and then return false.
```
Comment (thepinetree) [inline src/include/buffer/buffer_pool_manager_instance.h:132]:
```suggestion
   * page is pinned and cannot be deleted return false immediately.
```

Comment (thepinetree) [reply src/include/buffer/buffer_pool_manager_instance.h:132]: I'm not sure on this one, was trying to follow how code comments are described in other assignments.

File: src/include/buffer/buffer_pool_manager_instance.h [hunk 5]
```diff
@@ -98,65 +114,32 @@ class BufferPoolManagerInstance : public BufferPoolManager {
...
    *
-   * 1. Delete the page from the page table.
-   * 2. Remove the frame from the replacer.
-   * 3. Add the frame back to the free list.
-   * 4. Reset the page's metadata (id, memory, dirty flag).
-   * 4. Call DeallocatePage() to imitate freeing the page on disk.
+   * After deleting the page from the page table, remove the tracking of the frame from the replacer and add the frame
```
Comment (thepinetree) [inline src/include/buffer/buffer_pool_manager_instance.h:134]:
```suggestion
   * After deleting the page from the page table, stop tracking the frame in the replacer and add the frame
```

**Review by thepinetree — APPROVED** [review]
Minor nits, again feel free to ignore, ty for changes!

**Pull request merged**